mlflow/utils/autologging_utils/safety.py

```diff
@@ -310,6 +310,26 @@ class definition or a function object. If it is a function object, the
     else:
         assert callable(patch_function)
 
+    original_fn = gorilla.get_original_attribute(
+        destination, function_name, bypass_descriptor_protocol=False
+    )
+    # Retrieve raw attribute while bypassing the descriptor protocol
+    raw_original_obj = gorilla.get_original_attribute(
+        destination, function_name, bypass_descriptor_protocol=True
+    )
+    if original_fn != raw_original_obj:
+        raise RuntimeError(f"Unsupport patch on {str(destination)}.{str}")
```

Comment:
Can we replace this with:

```
try:
    original_fn = gorilla.get_original_attribute(
        destination, function_name, bypass_descriptor_protocol=True
    )
except AttributeError:
    raise MlflowException("Unsupported patch on {str(destination)}.{function_name}. The specified function must be defined and accessible on the specified object without relying on the descriptor protocol")
```

I don't think we need to define both `original_fn` and `raw_original_obj` if we're asserting that they're equal.

Reply:
^ Perhaps this doesn't work for data descriptors, since data descriptors take precedence over attributes in the object's `__dict__`? Let me know!

Regardless, can we refer to `original_fn` rather than `raw_original_fn` throughout the rest of this method?